src/app/marathon/schedule-management/edit/edit.component.ts

```diff
@@ -77,10 +231,292 @@ export class EditComponent implements OnInit {
+        ...availabilities,
+      };
+
+      for (const [key, value] of Object.entries(availabilities)) {
+
+        // in case of adding a runner that did not submit.
+        // TODO: add to empty array either way?
+        if (!value.length) {
+          continue;
+        }
+
+        // data already exists, just ignore
+        if (this.availabilitiesGroups.get(key)) {
+          continue;
+        }
+
+        this.availabilitiesGroups.add({
+          id: key,
+          content: value[0].username,
+        });
+
+        value.forEach((availability, index) => {
+          this.availabilitiesItems.add({
+            id: key + index,
+            group: key,
+            start: availability.from,
+            end: availability.to,
+            content: '',
+          });
+        });
+      }
+    } catch (e: any) {
+      console.error(e);
+    }
+  }
+
+  clearSelectedAvailabilities() {
+    const toRemove = this.availabilitiesItems.getIds({
+      filter: (item) => this.availabilitiesSelected.includes(item.group),
+    });
+    this.availabilitiesSelected = [];
+    this.availabilitiesSelectedItems = this.availabilitiesSelectedItems.filter(el => !toRemove.includes(el));
+    this.timeline.setSelection(this.availabilitiesSelectedItems);
+  }
+
+  triggerSelectedAvailability(inp: { username: string, on: boolean }) {
+    if (inp.on) {
+      this.selectAvailabilities(inp.username);
+    } else {
+      this.unselectAvailabilities(inp.username);
+    }
+  }
+
+  selectAvailabilities(username: string) {
+    this.availabilitiesSelected.push(username);
+    this.availabilitiesSelectedItems = [
+      ...new Set([
+        ...this.availabilitiesSelectedItems,
+        ...this.availabilitiesItems.getIds({
+          filter: (item) => item.group === username,
+        }),
+      ]),
+    ];
+    this.timeline.setSelection(this.availabilitiesSelectedItems);
+  }
+
+  unselectAvailabilities(username: string) {
+    this.availabilitiesSelected.splice(this.availabilitiesSelected.findIndex(name => name === username), 1);
+    const toRemove = this.availabilitiesItems.getIds({
+      filter: (item) => item.group === username,
+    });
+    this.availabilitiesSelectedItems = this.availabilitiesSelectedItems.filter(el => !toRemove.includes(el));
+    this.timeline.setSelection(this.availabilitiesSelectedItems);
+  }
+
+  addCustomLine(isSetup: boolean): void {
+    const customLine: V2ScheduleLine = {
+      category: '',
+      categoryId: 0,
+      console: '',
+      customData: '',
+      customRun: true,
+      date: undefined,
+      emulated: false,
+      estimate: 'PT0S',
+      game: '',
+      id: -1,
+      position: this.lines.length,
+      ratio: '',
+      runners: [],
+      setupBlock: isSetup,
+      setupBlockText: '',
+      setupTime: this.marathonService.marathon.defaultSetupTime,
+      type: 'OTHER',
+    };
+
+    this.lines.push(customLine);
+
+    this.scheduleTable.toggleExpand(this.lines.length - 1);
```

Comment:
could potentially use the helper here and just pass a boolean to always force it to be open